packages/core/src/i18n.ts

```diff
@@ -7,23 +7,26 @@ import { EventEmitter } from "./eventEmitter"
 export type MessageOptions = {
   message?: string
   context?: string
-  formats?: Object
+  formats?: Formats
 }
 
 export type Locale = string
 export type Locales = Locale | Locale[]
+export type Formats = Record<string, Intl.DateTimeFormatOptions | Intl.NumberFormatOptions>
+
+export type Values = Record<string, unknown>;
 
 export type LocaleData = {
-  plurals?: Function
+  plurals?: (n: number, ordinal?: boolean) => number
```

Comment:
The return type needs to be `PluralCategory` from the `make-plural` instead of `number`.